AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs

```diff
@@ -231,14 +231,32 @@ public bool AutoStartUpload
 
         protected override void OnInit(EventArgs e) {
             base.OnInit(e);
-            if(!IsDesignMode) {
-                if(!string.IsNullOrEmpty(Page.Request.QueryString["contextkey"])
+            if(IsDesignMode || !IsControlRequest()) 
+                return;
+                
+            IsInFileUploadPostBack = true;
+
+            var processor = new UploadRequestProcessor {
+                Context = Context,               
+                UploadStart = UploadStart,
+                UploadComplete = UploadComplete,
+                UploadCompleteAll = UploadCompleteAll,
+                SetUploadedFilePath = SetUploadedFilePath
+            };
+
+            processor.ProcessRequest();
+        }
+
+        bool IsControlRequest() {
```

Comment:
`IsControlRequest` - need better name. It should be connected to `IsInFileUploadPostBack`.
Maybe `HasFileUploadPostBackParams`